### PR TITLE
Add Swinject as an internal dependency

### PR DESCRIPTION
**Pull request by skorulis-ap**
This copies the sources and tests from Swinject and SwinjectAutoRegistration into Knit to remove the external dependency. This will allow us to merge in the changes we want without the risk of breaking projects which depend on Swinject

### CODE REVIEW

File: Sources/KnitCodeGen/ConfigurationSet.swift [hunk 1]
```diff
@@ -62,7 +62,7 @@ public extension ConfigurationSet {
 
     func makeTypeSafetySourceFile() throws -> String {
         var allImports = allImports
-        allImports.insert(.named("Swinject"))
+        allImports.insert(.named("Knit"))
```
Comment (skorulis-ap) [inline Sources/KnitCodeGen/ConfigurationSet.swift:65]: I changed this so external code no longer references Swinject.

File: Docs/Swinject.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Swinject
+
+[Swinject](https://github.com/Swinject/Swinject/) and [SwinjectAutoregistration](https://github.com/Swinject/SwinjectAutoregistration) have been copied into the Knit project in order to allow deviation from the Swinject project which is not being well maintained.
```
Comment (bradfol) [inline Docs/Swinject.md:3]: Can we include the source commits from those projects?

File: Sources/Swinject/Info.plist
Comment (bradfol) [inline Sources/Swinject/Info.plist:1]: Needed?

File: Sources/Swinject/UnavailableItems.swift
Comment (bradfol) [inline Sources/Swinject/UnavailableItems.swift:1]: Probably don't need this

File: Sources/SwinjectAutoregistration/Info.plist
Comment (bradfol) [inline Sources/SwinjectAutoregistration/Info.plist:1]: Needed?

File: Sources/SwinjectAutoregistration/SwinjectAutoregistration.h
Comment (bradfol) [inline Sources/SwinjectAutoregistration/SwinjectAutoregistration.h:1]: Remove?

File: Tests/SwinjectTests/Info.plist
Comment (bradfol) [inline Tests/SwinjectTests/Info.plist:1]: Another info.plist

**Review by bradfol — APPROVED** [review]
🎉

File: Docs/Swinject.md (hunk 2 above)
Comment (skorulis-ap) [reply Docs/Swinject.md:3]: Good idea, listed the source commits below.

File: Sources/Swinject/Info.plist
Comment (skorulis-ap) [reply Sources/Swinject/Info.plist:1]: I took out all of the file that we don't need.

**Review by brianpartridge — APPROVED** [review]

**Pull request merged**